### PR TITLE
feat: slot items for Header component (react)

### PR DESCRIPTION
**Pull request by ramonmulia**
Add Slot items for the Header component

- header slot feature
- tests
- examples how it could be used (storybook)
- breakpoints fixed (mobile and tablet)

Screenshot 2024-12-04 at 07 42 41

Screenshot 2024-12-04 at 07 43 17

### CODE REVIEW

File: packages/react/ds/src/header/components/header-menu.tsx [hunk 1]
```diff
@@ -16,19 +19,97 @@ type HeaderMenuProps = {
...
+  item: { label?: string; slot: React.ReactNode };
+};
+
+const MenuListItem = ({ href = '#', label = '', bold = true }) => (
+  <a
+    aria-label={label || 'link with no label'}
+    href={href}
+    className="hover:gi-underline hover:gi-underline-offset-sm gi-block gi-py-4 gi-border-gray-100 gi-border-b-xs gi-border-solid"
+  >
+    <span className={cn('gi-text-sm', 'gi-ml-1', { 'gi-font-bold': bold })}>
```
Comment (BogdanDanilescu) [inline packages/react/ds/src/header/components/header-menu.tsx:35]: do we need `{ 'gi-font-bold': bold }` ? can't we do just `gi-font-bold` instead ?

File: packages/react/ds/src/header/components/header-menu.tsx [hunk 2]
```diff
@@ -16,19 +19,97 @@ type HeaderMenuProps = {
...
     icon?: IconId;
   };
+} & Pick<HeaderProps, 'tools'>;
+
+type MenuItemAccordionProps = {
+  index: number;
+  item: { label?: string; slot: React.ReactNode };
+};
+
+const MenuListItem = ({ href = '#', label = '', bold = true }) => (
```
Comment (BogdanDanilescu) [inline packages/react/ds/src/header/components/header-menu.tsx:29]: can we add types to this function ?

File: packages/react/ds/src/header/components/header-slot.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import { useEffect } from 'react';
+import ReactDOMClient from 'react-dom/client';
+import { Icon } from '../../icon/icon.js';
+
+type SloTProps = {
```
Comment (BogdanDanilescu) [inline packages/react/ds/src/header/components/header-slot.tsx:5]: the letter 'T' should be lowercase

File: packages/react/ds/src/header/components/header-slot.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  return null;
+};
+
+const SlotItemAction = ({ item: { label, icon }, index }: SloTProps) => {
+  return (
+    <label
+      htmlFor={`ItemActionTrigger-${index}`}
+      className="gi-header-tool-item"
+    >
+      <input
```
Comment (BogdanDanilescu) [inline packages/react/ds/src/header/components/header-slot.tsx:53]: is it a good idea to use the `Checkbox` component here ?

File: packages/react/ds/src/header/components/header-menu.tsx [hunk 5]
```diff
@@ -16,19 +19,107 @@ type HeaderMenuProps = {
...
+    href={href}
+    className="hover:gi-underline hover:gi-underline-offset-sm gi-block gi-py-4 gi-border-gray-100 gi-border-b-xs gi-border-solid"
+  >
+    <span className={cn('gi-text-sm', 'gi-ml-1', { 'gi-font-bold': bold })}>
+      {label}
+    </span>
+  </a>
+);
+
+export const MenuItemAccordion = ({ index, item }: MenuItemAccordionProps) => {
```
Comment (pregno) [inline packages/react/ds/src/header/components/header-menu.tsx:51]: Why an `accordion`?

File: packages/react/ds/src/header/components/header-search.tsx [hunk 6]
```diff
@@ -14,28 +14,23 @@ function HeaderSearch({
...
-            className="gi-flex-auto"
-          />
-          <Button className="gi-ml-1 gi-flex-none">
-            Search
-            <Icon icon={icon} ariaLabel="Search" />
-          </Button>
-        </div>
-      </form>
-    </div>
+    <form action={ActionType} className="gi-max-w-md gi-mx-auto">
```
Comment (pregno) [inline packages/react/ds/src/header/components/header-search.tsx:17]: There is probably something wrong with the merge.
The current version of `main` has the correct styling.

File: packages/react/ds/src/stack/stack.tsx [hunk 7]
```diff
@@ -204,6 +204,7 @@ export const Stack: React.FC<StackProps> = ({
     'gi-flex',
     'gi-overflow-auto',
     'gi-w-full',
+    'gi-p-1',
```
Comment (pregno) [inline packages/react/ds/src/stack/stack.tsx:207]: Why this padding?

File: packages/react/ds/src/header/components/header-slot.tsx (hunk 4 above)
Comment (pregno) [reply packages/react/ds/src/header/components/header-slot.tsx:53]: If you are using the `checkbox` why do you need to attach the listener to the buttons with `attachEventsToItemActionTriggers`?

Comment (ramonmulia) [reply packages/react/ds/src/header/components/header-slot.tsx:53]: I needed to attach events on the buttons to toggle the icons, hide all slots opened (including search) and show the current one.
As we have differents custom items slots it weren't possible to have css rules for that (like SearchTrigger), so attaching the event `change` will allow the all checkboxs close and open just the one clicked.

File: packages/react/ds/src/header/components/header-menu.tsx (hunk 5 above)
Comment (ramonmulia) [reply packages/react/ds/src/header/components/header-menu.tsx:51]: Afeter some discussions with Lexa we came out with this solution, just for mobile in this case: https://www.figma.com/design/ZEqfpXhocdZs1K5YfsE5ws/Design-system?node-id=3412-1644&node-type=canvas&m=dev

this way if the user wants, the user can include the prop `keepOnMobile` and if it is true we show the slots in mobile/tablet

File: packages/react/ds/src/stack/stack.tsx (hunk 7 above)
Comment (ramonmulia) [reply packages/react/ds/src/stack/stack.tsx:207]: my bad, going to remove it.

File: packages/react/ds/src/header/components/header-search.tsx (hunk 6 above)
Comment (ramonmulia) [reply packages/react/ds/src/header/components/header-search.tsx:17]: The styles are the same, I just moved the div container to the parent component, so in that case we show the right styles only on desktop.

**Review by pregno — APPROVED** [review]

File: packages/react/ds/src/header/components/header-menu.tsx [hunk 8]
```diff
@@ -16,19 +19,107 @@ type HeaderMenuProps = {
...
+  return (
+    <div
+      key={`Accordion-item-${index}`}
+      id={`Accordion-item-${index}`}
+      className="gi-accordion-item-container"
+      data-open="false"
+    >
+      <div
+        aria-label={item.label}
+        className="gi-accordion-item-toggle hover:gi-no-underline hover:gi-underline-offset-sm gi-block gi-py-4 gi-border-gray-100 gi-border-b-xs gi-border-solid"
```
Comment (hamza14khan) [inline packages/react/ds/src/header/components/header-menu.tsx:92]: question: Can we create a class for this?

File: packages/react/ds/src/header/components/header-menu.tsx [hunk 9]
```diff
@@ -16,19 +19,107 @@ type HeaderMenuProps = {
...
+      >
+        <div className="gi-full-width gi-flex gi-justify-between">
+          <span className="gi-text-sm gi-font-bold gi-ml-1">{item.label}</span>
+          <Icon icon="chevron_right" className="gi-accordion-item-icon" />
+        </div>
+      </div>
+
+      <div className="gi-accordion-item-slot gi-hidden gi-bg-gray-50 gi-p-4">
```
Comment (hamza14khan) [inline packages/react/ds/src/header/components/header-menu.tsx:100]: same as above.

**Review by BogdanDanilescu — APPROVED** [review]

**Pull request merged**